src/nodes/model/custom/Custom.ts

```diff
@@ -15,36 +15,37 @@ export default class Custom extends Node {
 
   constructor() {
     super();
-    this.addOption('Enter Func', 'CodeVaultButtonOption', undefined, undefined, { customNode: this });
+    this.addOption('Select Function', 'CodeVaultButtonOption', undefined, undefined, { customNode: this });
   }
 
   public load(state: INodeState) {
     super.load(state);
     this.updateNode();
   }
 
-  public setInlineCode(inlineCode: string) {
+  public setInlineCode(inlineCode?: ParsedFunction) {
     this.state.inlineCode = inlineCode;
     this.updateNode();
   }
 
-  public getInlineCode() {
-    return this.state.inlineCode ? this.state.inlineCode : '';
+  public getInlineCode(): (ParsedFunction | undefined) {
+    const { inlineCode } = this.state;
+    if (this.state.inlineCode) {
+      return new ParsedFunction(inlineCode.name, inlineCode.body, inlineCode.args);
+    }
+    return undefined;
   }
 
   private updateNode() {
-    if (this.getInlineCode() === '') {
+    const inlineCode = this.getInlineCode();
+    if (!inlineCode) {
       this.name = Nodes.Custom;
       this.removeAllInputs();
```

Comment:
I know that this isn't relating to the changes that you've made in this PR, but how hard would it be to not call `this.removeAllInputs()`, and instead only add ones which have been changed? The problem is that when I link up a custom node with an input and output, but then whilst it's still attached to other nodes I change the code to have an additional parameter or whatever, all the connections get deleted... The output connection does weird stuff too.

Reply:
Yes you're right, we should support that! Maybe for next checkpoint though? I'll create a ticket and see what I can do!

Reply:
Alrighty, I'll approve then